regression?: jbang jdk install of specific jdk

got this from @gsmet:

jbang blocking quarkus release

```
[jbang] [2026-06-24 11:31:36] Using JDK provider(s): CurrentJdkProvider, DefaultJdkProvider, JavaHomeJdkProvider, PathJdkProvider, LinkedJdkProvider, JBangJdkProvider
Error:  [0:219] [ERROR] When providing an existing JDK path, the versionOrId parameter must be a non-integer id
java.lang.IllegalArgumentException: When providing an existing JDK path, the versionOrId parameter must be a non-integer id
    at dev.jbang.cli.Jdk$JdkInstall.doCall(Jdk.java:123)
    at dev.jbang.cli.BaseCommand.execute(BaseCommand.java:134)
    at org.aesh.command.impl.operator.EndOperator.execute(EndOperator.java:41)
    at org.aesh.command.impl.Executions$ExecutionImpl.execute(Executions.java:239)
    at org.aesh.command.impl.AeshCommandRuntime.runExecutor(AeshCommandRuntime.java:222)
    at org.aesh.command.impl.AeshCommandRuntime.executeCommand(AeshCommandRuntime.java:175)
    at org.aesh.AeshRuntimeRunner.execute(AeshRuntimeRunner.java:160)
    at dev.jbang.Main.main(Main.java:46)
```

I was doing:

> jbang --verbose jdk install java-21 ${JAVA_HOME_21_X64}
> jbang --java 21 ...

So that JBang doesn't download Java 21

 I tried to switch to using a non numeric idGuillaume Smet: but --java only supports numeric ids and I don't see any other parameter to push an install id

--- a/src/main/java/dev/jbang/cli/Jdk.java
+++ b/src/main/java/dev/jbang/cli/Jdk.java
@@ -120,8 +120,8 @@ public class Jdk extends BaseCommand {
 				}
 				if (!Util.isNullOrBlankString(path)) {
 					if (isValidInteger(versionOrId)) {
-						throw new IllegalArgumentException(
-								"When providing an existing JDK path, the versionOrId parameter must be a non-integer id");
+						versionOrId = generateUserId(versionOrId);
+						Util.infoMsg("Numeric id detected, using '" + versionOrId + "' as id");
 					}
 					Path jdkCacheDir = Settings.getCacheDir(Cache.CacheClass.jdks);
 					Path jdkPath = Paths.get(path);
@@ -150,6 +150,10 @@ public class Jdk extends BaseCommand {
 			} catch (NumberFormatException e) {
 				return false;
 			}
+		}
+
+		private String generateUserId(String version) {
+			return version + "-user";
 		}
 	}
 

--- a/src/test/java/dev/jbang/cli/TestJdk.java
+++ b/src/test/java/dev/jbang/cli/TestJdk.java
@@ -350,10 +350,20 @@ public class TestJdk extends BaseTest {
 	}
 
 	@Test
-	void testJdkInstallWithLinkingAndIntegerId() {
-		assertThrows(IllegalArgumentException.class,
-				() -> checkedRun("install", "--force", "11", "/non-existent-path"),
-				"When providing an existing JDK path, the versionOrId parameter must be a non-integer id");
+	void testJdkInstallWithLinkingAndIntegerIdGeneratesUserId(
+			@TempDir File javaDir) throws Exception {
+		initMockJdkDir(javaDir.toPath(), "11.0.14");
+		final Path jdkPath = Settings.getCacheDir(Cache.CacheClass.jdks);
+
+		CaptureResult<Integer> result = checkedRun("install", "--force", "11", javaDir.toPath().toString());
+
+		assertThat(result.result, equalTo(SUCCESS_EXIT));
+		assertThat(result.normalizedErr(),
+				containsString("Numeric id detected, using '11-user' as id"));
+		assertThat(result.normalizedErr(),
+				containsString("JDK 11-user-linked has been linked to: " + javaDir.toPath().toString()));
+		assertTrue(Util.isLink(jdkPath.resolve("11-user-linked")));
+		assertTrue(Files.isSameFile(javaDir.toPath(), jdkPath.resolve("11-user-linked").toRealPath()));
 	}
 
 	@Test
@@ -367,12 +377,18 @@ public class TestJdk extends BaseTest {
 	}
 
 	@Test
-	void testJdkInstallWithLinkingToExistingJdkPathWithDifferentVersion(@TempDir File javaDir) {
+	void testJdkInstallWithLinkingToExistingJdkPathWithDifferentVersion(
+			@TempDir File javaDir) throws Exception {
 		initMockJdkDir(javaDir.toPath(), "11.0.14");
+		final Path jdkPath = Settings.getCacheDir(Cache.CacheClass.jdks);
 
-		assertThrows(Exception.class,
-				() -> checkedRun(withProviders(
-						"jdk", "install", "--force", "13", javaDir.toPath().toString())));
+		CaptureResult<Integer> result = checkedRun(withProviders(
+				"jdk", "install", "--force", "13", javaDir.toPath().toString()));
+
+		assertThat(result.result, equalTo(SUCCESS_EXIT));
+		assertThat(result.normalizedErr(),
+				containsString("Numeric id detected, using '13-user' as id"));
+		assertTrue(Util.isLink(jdkPath.resolve("13-user-linked")));
 	}
 
 	@Test

